packages/@ember/-internals/container/lib/container.ts

```diff
@@ -143,7 +143,7 @@ export default class Container {
    @param {String} [options.source] The fullname of the request source (used for local lookup)
    @return {any}
    */
-  lookup(fullName: string, options: LookupOptions): any {
+  lookup<T>(fullName: string, options?: TypeOptions): T | undefined {
```

Comment:
This should actually return `unknown`, since it's impossible to infer from the arguments what the return type `T` would actually be.

```suggestion
  lookup(fullName: string, options?: TypeOptions): unknown {
```

I know this matches the *other* implementations, but we should take care to move *toward* rather than *away from* safety with this work where our hands are not already tied in some way.

I expect you introduced this because there is client code which is relying on being able to freely cast because of the previous `any` return type?

Reply:
This makes sense. I’ll give it a second look tomorrow.